src/Data/Parameterized/List.hs

```diff
@@ -267,10 +275,10 @@ deriving instance Show  (Index l x)
 
 instance ShowF (Index l)
 
-instance TestEquality (Index l) where
-  testEquality IndexHere IndexHere = Just Refl
-  testEquality (IndexThere x) (IndexThere y) = testEquality x y
-  testEquality _ _ = Nothing
```

Comment:
I presume that this instance was removed because it does not obey the `TestEquality` laws? If so, should we remove the instances for other `Index` data types in `parameterized-utils` (e.g., `Data.Parameterized.Context.{Safe,Unsafe}.Index`)? Or would it be better to keep this `TestEquality` instance for a while to give users a chance to migrate to `EqF`?

Reply:
Yeah, and I actually meant to put this back in to allow just such a gradual change.

Initially I had actually removed the instances in `Context`, but realized that it might screw people up in the short term if I did that.

Do you know if there is a way to deprecate a type class instance and have GHC give a warning that they should use `eqF` instead of `testEquality`?

Reply:
Yep, this would entail a lot of churn -- not only will we uncover missing `EqF` instances downstream, this will invalidate all existing `EqF` instances (although I'm not sure if there are too many of those).

But yeah, the solution should just be to copy-paste instances.

I made this PR in discussion with RobD, as a way to start the conversation about whether this change is worth it. I think it is worth it, personally -- insofar as `parameterized-utils` is Galois' offering of a "standard library for dependent types in Haskell", we want to be in compliance with all the basic laws.